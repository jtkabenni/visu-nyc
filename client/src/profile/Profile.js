import React, { useContext } from "react";
import UserContext from "../auth/UserContext";

import SavedMapList from "../maps/SavedMapList";
import { Link } from "react-router-dom";

function Profile() {
  const { user } = useContext(UserContext);
  console.log(user);
  return (
    <div class="container">
      {user ? (
        <div class="profile-sections">
          <div class="profile-details">
            <h3>
              {user.firstName} {user.lastName}
            </h3>
            <div class="profile-detail">
              <p>
                <b>Username</b>
              </p>
              <p> {user.username}</p>
            </div>
            <div class="profile-detail">
              <p>
                <b>Email address</b>
              </p>
              <p> {user.email}</p>
            </div>
            <div class="profile-detail">
              <p>
                <b>Total maps</b>
              </p>
              <p> {user.maps.length} maps</p>
            </div>

            <button>
              <Link to="/profile/update">Update</Link>
            </button>
          </div>

          <SavedMapList maps={user.maps}></SavedMapList>
        </div>
      ) : (
        ""
      )}
    </div>
  );
}
export default Profile;
