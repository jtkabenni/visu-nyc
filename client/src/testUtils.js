import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  is_admin: true,
  auth_type: "custom",
  maps: [],
};

const demoCurrUserMaps = [];

const UserProvider = ({
  children,
  currUser = demoUser,
  currUserMaps = demoCurrUserMaps,
}) => (
  <UserContext.Provider value={{ currUser, currUserMaps }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider };
