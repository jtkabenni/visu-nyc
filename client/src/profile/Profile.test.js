import React from "react";
import { render } from "@testing-library/react";
import Profile from "./Profile";
import { UserProvider } from "../testUtils";
import { BrowserRouter as Router } from "react-router-dom";

it("matches snapshot", function () {
  const { asFragment } = render(
    <Router>
      <UserProvider>
        <Profile />
      </UserProvider>
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});
