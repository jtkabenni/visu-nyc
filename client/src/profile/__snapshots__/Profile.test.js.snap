// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="profile-sections"
    >
      <div
        class="profile-details"
      >
        <h3>
           
        </h3>
        <div
          class="profile-detail"
        >
          <p>
            <b>
              Username
            </b>
          </p>
          <p>
             testuser
          </p>
        </div>
        <div
          class="profile-detail"
        >
          <p>
            <b>
              Email address
            </b>
          </p>
          <p>
             test@test.net
          </p>
        </div>
        <div
          class="profile-detail"
        >
          <p>
            <b>
              Total maps
            </b>
          </p>
          <p>
             0 maps
          </p>
        </div>
        <button>
          <a
            href="/profile/update"
          >
            Update
          </a>
        </button>
      </div>
      <div
        class="map-list"
      >
        <h3>
          All maps
        </h3>
        <i>
          No maps added yet.
        </i>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
